wordpress-mobile/WordPress-Android
# [Pages] Implement virtual Homepage actions

Part of #18371 

This implements the missing part of the #18400 PR, which are the callbacks and actions to:
- Open the SiteEditor on a WebView (injecting CSS to hide some navigation elements of the page)
- Open the support page about `Templates` on the external browser

This also implements some code to inject CSS to the page being loaded in the WebView.

## To test

Requirements:
- At least 1 site using a block-based theme, such as `Vetro`, with the homepage set to display "your latest posts" (on the web you can find it at `Appearance > Customize > Homepage Settings`

1. Login with your account
2. Select a site with a block-based theme
3. Go to `Debug Settings`
4. Enable the `SiteEditorMVPFeatureConfig`
5. Go back to the `My Site` screen
6. Tap on `Pages`
7. **Verify** the "virtual" homepage item is shown
8. Tap the card
9. **Verify** the Site Editor page is opened on a WebView directly on the editor for the home template/page
10. **Verify** UI for navigating inside the editor is hidden (compare with web version on a browser)
11. Go back to the `Pages` screen
12. Tap the info icon
13. **Verify** it opens the support page about `Templates` on the external browser

If possible, test with different types of accounts, ex: WP.com account, self-hosted without Jetpack connection, and self-hosted with Jetpack connection. And also with different site types, ex: Simple sites and Atomic sites. And also WordPress and Jetpack apps.

## Regression Notes
1. Potential unintended areas of impact
Issues for other page types, issues opening the editor/external url.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Added some unit tests in the ViewModel responsible for those callbacks.

PR submission checklist:


UI Changes testing checklist:
**There were no UI changes, just UI interaction, so most actions below don't apply**
